### DISCUSSION TITLE
Mécanisme "est applicable"

### DISCUSSION THREAD

**Discussion by mquandalle** [publicodes/publicodes#204] [Ideas]
Avant l'introduction du littéral `null` pour encoder une règle non applicable, on pouvait tester `règle = non` pour déterminer son applicabilité. Pour rappel, le problème de cette approche est qu'elle ne permet pas de distinguer la réponse négative à une question d'une règle désactivée :

```yaml
participation élection:
  applicable si: personne majeure
  question: Allez-vous voter aux élections ?
  type: booléen
```

Ici on veut pouvoir distinguer le cas `participation élection = non` (la règle s'applique et la réponse est négative) du cas `participation élection = null` (la règle ne s'applique pas)

Sauf que je suis un peu en manque d'inspiration sur une bonne manière d'exposer ce test dans publicodes :
- `x = null` ça fait trop “technique”
- `x = non applicable` c'est pas génial non plus
-  on pourrait avoir un mécanisme `est applicable: x` mais ça me semble aussi être quelque chose de compliqué à expliquer, surtout pour le distinguer de `applicable si:` et `rend non applicable`
- on pourrait avoir `x est applicable` est français, un peu comme `x is True` en python, mais le fait que notre grammaire permet aux variables de contenir des espaces rend la chose quasi impossible
- `x . est applicable = non` mais un peu bizarre de créer une fausse règle `. est applicable`
- modifier `variations: si:` et `applicable si` pour que dans le cas où `x` n'est pas un booléen, `si: x` signifie “si x est applicable” 

Des idées ?

**Comment by johangirod** [comment]
On a maintenant le mécanisme `est applicable: <valeur>` qui fait le taf'
